### PR TITLE
Add more countries to MC Tiler

### PR DESCRIPTION
**Pull request by antoniocarlon**
Added more countries to MC Tiler. Acceptance:
- [x] CA
- [x] AU
- [x] UK
  - [x] Levels 0-13
  - ~Level 14 (postcode units)~ See https://github.com/CartoDB/do_tiler/issues/43

~Blocked by https://github.com/CartoDB/observatory-extension/pull/354~

# Tilesets API MRLI Release

These PRs (together):
- CartoDB/bigmetadata/pull/578
- CartoDB/observatory-extension/pull/354
- CartoDB/do_tiler/pull/51

Close these issues:
- CartoDB/bigmetadata/issues/570
- CartoDB/bigmetadata/issues/583
- CartoDB/do_tiler/issues/45

# Merge warning

*Explanation about "do not merge yet" label*:
- This shouldn't be merged until a new DO release has happened.
- Remove the `FIXME`.

### CODE REVIEW

**Comment by antoniocarlon** [comment]
@juanignaciosl, I think that I have addressed your comments, another CR round?

**Comment by juanignaciosl** [comment]
Blocked until data is already loaded.

**Comment by juanignaciosl** [comment]
This is no longer blocked and can be accepted in staging (after fixing conflicts).

**Comment by juanignaciosl** [comment]
[etl_tiler.xyz.log.gz](https://github.com/CartoDB/bigmetadata/files/2531213/etl_tiler.xyz.log.gz)

After a couple of changes, including the one about valid geom, it keeps failing. I haven't been able to take a look at all, sorry.

**Comment by antoniocarlon** [comment]
@juanignaciosl I think it's working now (adding the `ST_MakeValid here: https://github.com/CartoDB/observatory-extension/pull/354)

**Comment by juanignaciosl** [comment]
`CA` is still runing (three days later!!!) U_U. It's not blocked but it's taking forever (a zoom 14 insert of dissemination areas)).

`UK` failed:

```
2018-11-05 17:52:39,606 [INFO]: [pid 1] Worker Worker(salt=768349666, workers=1, host=ffab29bb4f48, username=root, pid=1) running   tasks.tiler.xyz.AllSimpleDOXYZTables(config_file=uk_all.json)
2018-11-05 17:52:39,629 [ERROR]: [pid 1] Worker Worker(salt=768349666, workers=1, host=ffab29bb4f48, username=root, pid=1) failed    tasks.tiler.xyz.AllSimpleDOXYZTables(config_file=uk_all.json)
Traceback (most recent call last):
  File "/usr/local/lib/python3.5/dist-packages/luigi/worker.py", line 187, in run
    raise RuntimeError('Unfulfilled %s at run time: %s' % (deps, ', '.join(missing)))
RuntimeError: Unfulfilled dependencies at run time: tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_unit_14_7e71d201a5, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_sector_12_b1edd11dff, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_sector_13_6d131c226a, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_area_0_8ca9640990, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_area_1_d2e8b8cdc8, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_area_2_880488e448, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_area_3_69beff7af9, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_area_4_c5a93ea59f, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_area_5_35df67232a, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_area_6_e4ce64bcce, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_area_7_aefbe8cd1a, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_distric_8_64ac9b6b4b, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_distric_9_7627c65a7d, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_distric_10_e6f262e473, tasks.tiler.xyz.SimpleTilerDOXYZTableTask_uk_all_json_postcode_distric_11_13111928ab
2018-11-05 17:52:39,696 [INFO]: Informed scheduler that task   tasks.tiler.xyz.AllSimpleDOXYZTables_uk_all_json_9b04c8a962   has status   FAILED
```

**Comment by juanignaciosl** [comment]
Ok, UK failed because tiler tasks don't have the explicit dependency to DO schema. This will be addressed later. Meanwhile, running `uk-all` before should fix it.

**Comment by juanignaciosl** [comment]
This shouldn't be merged until a new DO release has happened.

File: tasks/ca/statcan/geo.py [hunk 1]
```diff
@@ -222,7 +250,10 @@ def columns(self):
 class Geography(TableTask):
 
     resolution = Parameter(default=GEO_PR)
-    year = IntParameter()
+    # FIXME: don't merge to master with this flag on. It's a workaround so
+    # CA can be run for the tiler without regenerating all the geographies.
+    # Once a new DO dump is deployed, this is not needed.
+    year = IntParameter(significant=False)
```
Comment (javitonino) [inline tasks/ca/statcan/geo.py]: We probably want to review the dump then, see what's missing and get it out of the way.

File: tasks/mc/data.py [hunk 2]
```diff
@@ -108,22 +108,57 @@ def geoname_format(country, name):
                        metro_level_name=METRO_LEVEL_NAME.get(country, 'metro'))
 
 
+HOST = 'http://172.17.0.1:8000/mc'
```
Comment (javitonino) [inline tasks/mc/data.py]: Maybe we want a parameter/`env` variable to overwrite this?

File: tasks/mc/data.py [hunk 3]
```diff
@@ -364,19 +413,28 @@ def _insert_tr(self, session):
...
+                    schema=output.schema,
+                    table=output.tablename,
                     region_id=REGION_ID_COLUMN[1],
                     month=MONTH_COLUMN[1],
                 )
-        session.execute(query)
+        try:
+            session.execute(query)
+        except Exception as e:
+            if re.match(self.PK_EXISTS_RE, str(e)):
```
Comment (javitonino) [inline tasks/mc/data.py]: Probably better to match by [error code](https://www.postgresql.org/docs/10/errcodes-appendix.html) which is accessible via `e.orig.pgcode` (`e` is SQLAlchemy's exception, `e.orig` is PsycoPG2's exception)

File: tasks/tiler/xyz.py [hunk 4]
```diff
@@ -14,21 +14,38 @@
 import csv
 
 LOGGER = get_logger(__name__)
+GEONAME_COLUMN = 'geoname'
+
+Z_RANGES = [
```
Comment (javitonino) [inline tasks/tiler/xyz.py]: Same ranges for all countries? Is this enough or do we need to be more flexible?

File: tasks/ca/statcan/geo.py (hunk 1 above)
Comment (juanignaciosl) [reply tasks/ca/statcan/geo.py]: I'm not sure about what you want to review. This change makes Tilesets API not having the latest geometries but the previous ones. Not a big deal. I'd remove this now, and rerun the XYZ tasks once we have a new DO dump with the latest CA geometries.

File: tasks/tiler/xyz.py (hunk 4 above)
Comment (juanignaciosl) [reply tasks/tiler/xyz.py]: It's just for partitioning. Currently, it's more than enough, there's no good reason for more fine-grained partitioning. I chose it based on the worst case, and it doesn't harm simpler cases.

**Comment by juanignaciosl** [comment]
Pushed changes. Sorry about the huge PR, it's been a long way :( In fact, you might want to take a look at `parent_names_for_geometries` first, as it's meant to be merged onto this one.

**Comment by juanignaciosl** [comment]
Blocked until a new DO dump with Australia 2016 and USA 2010, 2014 is deployed. Once it's ready, if there's no further plan for a new DO dump, a new dump, clean from scratch, will be generated with this branch (up-to-date with master, of course).

We're going to do it because this huge branch includes some ORM behavior changes that haven't been tested with the whole DO. So, before merging this into master, we want to check that it doesn't break the generation of the dump.

**Pull request merged**